### PR TITLE
Improve code

### PR DESCRIPTION
**Pull request by a2br**
Simplified and removed useless code.
- Made use of single-line `if`,
- `forEach` when `for` is useless,
- corrected some errors,
- simplified functions (unnecessary `return` => arrow functions),
- and used nullish operator `??=` to avoid redundant `if`/`else` statements.

### CODE REVIEW

File: src/data.js [hunk 1]
```diff
@@ -39,12 +39,9 @@ export const getAllData = async (list, all = false) => {
   const localData = localStorageHelper('get');
   const data = {};
 
-  if (localData && localData.date) {
```
Comment (a2br) [inline src/data.js:42]: No need to test `localData` if `localData.date` is tested

File: src/data.js [hunk 2]
```diff
@@ -55,14 +52,10 @@ export const getAllData = async (list, all = false) => {
 
     const statesData = {};
     const usData = {};
-    for (let i = 0; i < raw.length; i++) {
```
Comment (a2br) [inline src/data.js:58]: `i` isn't used elsewhere

**Review by a2br — COMMENTED** [review]
Please do not pay attention to this review, I misclicked when adding a comment.

File: src/data.js [hunk 3]
```diff
@@ -55,14 +52,10 @@ export const getAllData = async (list, all = false) => {
...
-      const dataPoint = raw[i];
+    raw.forEach((dataPoint) => {
 
-      if (statesData[dataPoint.state]) {
```
Comment (a2br) [inline src/data.js:61]: This can be simply be made with `??=`

File: src/data.js [hunk 4]
```diff
@@ -84,9 +78,9 @@ export const getAllData = async (list, all = false) => {
       statesData[name] = data.map((d, i) => {
         const current = { ...d };
         current.cases = Math.max(Number(current.cases) - totalCases, 0);
-        current.cases_accum = totalCases += current.cases;
```
Comment (a2br) [inline src/data.js:87]: Bad use of `+=`

File: src/summaryTable.js [hunk 5]
```diff
@@ -4,14 +4,11 @@ import * as d3 from 'd3';
...
-function getTotals(data) {
-  return data.reduce(
-    (res, day) => ({
-      cases: res.cases + day.cases,
-    }),
+const getTotals = (data) => data.reduce((res, day) => 
+    ({cases: res.cases + day.cases,}),
     { cases: 0 }
   );
-}
```
Comment (a2br) [inline src/summaryTable.js:14]: Changed this to an implicit `return` in arrow function

File: src/summaryTable.js [hunk 6]
```diff
@@ -22,13 +19,9 @@ export const initSummaryTable = (containerId) => {
   const table = d3.select(id).append('table').attr('class', 'summary-table');
 };
 
-function abbreviateNumber(value) {
-  return d3.format(".3s")(value);
-}
```
Comment (a2br) [inline src/summaryTable.js:27]: Changed this to an implicit `return` in arrow function

File: src/summaryTable.js [hunk 7]
```diff
@@ -22,13 +19,9 @@ export const initSummaryTable = (containerId) => {
...
+const abbreviateNumber = (value) => d3.format(".3s")(value);
 
-function formatNumberWithCommas(value) {
-  return d3.format(",r")(value);
-}
```
Comment (a2br) [inline src/summaryTable.js:31]: Changed this to an implicit `return` in arrow function

**Comment by a2br** [comment]
Oops, what a mess! I thought I could simply comment on my changes in the 'Files changed' section, but it appears it posts my comments here. Sorry, I'm not very familiar with this part of GitHub.

File: src/data.js (hunk 1 above)
Comment (mahowa) [reply src/data.js:42]: if localData is null localData.date will throw an error. So you do need this

Comment (mahowa) [reply src/data.js:42]: you could change it to localData?.date (i think this project allows optional chaining)

**Comment by mahowa** [comment]
If you could fix that one issue i will merge

File: src/data.js (hunk 1 above)
Comment (a2br) [reply src/data.js:42]: Corrected 👍

**Review by mahowa — APPROVED** [review]

**Pull request merged**